src/main/scala/edu/ie3/simona/agent/participant/hp/HpAgentFundamentals.scala

```diff
@@ -261,8 +261,26 @@ trait HpAgentFundamentals
       currentTick,
       updatedState,
     )
-    val updatedBaseStateData =
-      baseStateData.copy(stateDataStore = updatedStateDataStore)
+
+    val updatedBaseStateData = {
+      updatedState.maybeThermalThreshold match {
+        case Some(nextThreshold)
+            if baseStateData.foreseenDataTicks.headOption.exists {
+              case (_, Some(tick)) => nextThreshold.tick < tick
+              case _               => false
+            } =>
+          baseStateData.copy(
+            stateDataStore = updatedStateDataStore,
+            additionalActivationTicks =
+              baseStateData.additionalActivationTicks + nextThreshold.tick,
```

Comment:
`+` is deprecated, this is not:

```suggestion
              baseStateData.additionalActivationTicks ++ Set(nextThreshold.tick),
```